src/plugins/discover/public/application/angular/context/query_parameters/actions.ts

```diff
@@ -6,35 +6,47 @@
  * Side Public License, v 1.
  */
 
-import _ from 'lodash';
-import { esFilters } from '../../../../../../data/public';
-import { popularizeField } from '../../../helpers/popularize_field';
+import { pick } from 'lodash';
 
+import { esFilters, Filter, IFieldType } from '../../../../../../data/public';
+import { FilterManager, IndexPatternsContract } from '../../../../../../data/public';
+import { popularizeField } from '../../../helpers/popularize_field';
+import { ContextAppState, QueryParameters } from '../../context_app_state';
 import { MAX_CONTEXT_SIZE, MIN_CONTEXT_SIZE, QUERY_PARAMETER_KEYS } from './constants';
 
-export function getQueryParameterActions(filterManager, indexPatterns) {
-  const setPredecessorCount = (state) => (predecessorCount) =>
-    (state.queryParameters.predecessorCount = clamp(
+export function getQueryParameterActions(
+  filterManager: FilterManager,
+  indexPatterns?: IndexPatternsContract
+) {
+  const setPredecessorCount = (state: ContextAppState) => (predecessorCount: number) => {
+    return (state.queryParameters.predecessorCount = clamp(
       MIN_CONTEXT_SIZE,
       MAX_CONTEXT_SIZE,
       predecessorCount
     ));
+  };
 
-  const setSuccessorCount = (state) => (successorCount) =>
-    (state.queryParameters.successorCount = clamp(
+  const setSuccessorCount = (state: ContextAppState) => (successorCount: number) => {
+    return (state.queryParameters.successorCount = clamp(
       MIN_CONTEXT_SIZE,
       MAX_CONTEXT_SIZE,
       successorCount
     ));
+  };
 
-  const setQueryParameters = (state) => (queryParameters) =>
-    Object.assign(state.queryParameters, _.pick(queryParameters, QUERY_PARAMETER_KEYS));
+  const setQueryParameters = (state: ContextAppState) => (queryParameters: QueryParameters) => {
+    return Object.assign(state.queryParameters, pick(queryParameters, QUERY_PARAMETER_KEYS));
+  };
 
-  const updateFilters = () => (filters) => {
+  const updateFilters = () => (filters: Filter[]) => {
     filterManager.setFilters(filters);
   };
 
-  const addFilter = (state) => async (field, values, operation) => {
+  const addFilter = (state: ContextAppState) => async (
+    field: string | IFieldType,
+    values: any,
```

Comment:
:art: Let's use `unknown` instead of `any` here. Our styleguide is trying to emphasize on avoiding `any` whenever possible: https://github.com/elastic/kibana/blob/master/STYLEGUIDE.md#avoid-any-whenever-possible In this specific case it's even not needed since we don't do anything with that value, just passing it to a method that expects an `any` parameter, so we can perfectly use `unknown` here.

Reply:
Solved this particular argument type. Remaining `any`type occurrences are left in the some legacy test files